src/backend/stream.rs

```diff
@@ -646,14 +646,12 @@ impl<'ctx> StreamOps for PulseStream<'ctx> {
         match self.input_stream {
             None => Err(Error::error()),
             Some(ref stm) => match stm.get_latency() {
-                Ok(StreamLatency::Positive(w_usec)) => {
+                  Ok(StreamLatency::Positive(w_usec))
+                | Ok(StreamLatency::Negative(w_usec)) => {
```

Comment:
@kinetiknz can you please tell us your opinion here. With that change a negative latency is reported as normal latency. According to the doc negative latency can exist:

"In case the stream is a monitoring stream the result can be negative, i.e. the captured samples are not yet played."

It must be a quirk that applies only to capture. Do you see any problem with that?

Reply:
If I understand correctly, in the negative latency case PA is the difference between the end of the capture buffer (i.e. the latest available position) and where we've read from via peek/drop so far. If that's the case, I'd expect negative latency (or zero) is the normal case for capture streams.

I don't think we should be treating positive and negative the same. Do we ever see positive latency on capture streams?

Reply:
I am not 100% sure what the negative latency is. I was testing it, the common case was the positive latency but once in a while it was given a negative latency and it was panicking (previous version of the patch).

Reply:
Apologies, I ran out of time today. I'll post a better explanation of my understand tomorrow.

Reply:
I link the documentation [here](https://freedesktop.org/software/pulseaudio/doxygen/stream_8h.html#aa521efcc16fe2abf0f8461462432ac16) for reference.

Reply:
I wonder if this "monitoring stream" mentioned in the documentation are streams that a monitor output device is attached as an input device. I am testing with the monitor device and a negative latency is always returned. This behavior is different than what I was observing in the previous test. Then, the problem was sporadic and a negative latency was returned after a few tries. I wish I had checked the devices involved. Now, I never have a negative latency with a non-monitor device and I always have on a monitor device.

Reply:
I discussed it with @ford-prefect in IRC. The bottom line was that:

- A monitor stream is a record stream that has a monitor source device attached.
- Negative latency is possible only in monitor streams
- A normal record stream cannot have a negative latency.

Arun's advice was to continue to assert/panic on normal record streams that the latency is never negative and to truncate the latency to zero when it is negative in monitor record streams. We can check if a stream is a monitor one by [pa_source_info](https://freedesktop.org/software/pulseaudio/doxygen/structpa__source__info.html#a5e304b796ce71c7fa54e5a88f3333630).

Reply:
Thanks for following this up.  I don't think there's much value in treating monitor streams differently here, so we can simply treat positive latency values as usual and clamp negative latency values to zero.  It'd be best to add a comment to the code explaining this, also.

Reply:
In this case, the last patch is what we want. @Koalab99 can you add a comment to clarify why the negative latency is truncated to zero according to what we have discussed above?